### DISCUSSION TITLE
Logging inside C code

### DISCUSSION THREAD

**Discussion by krishnak** [micropython/micropython#9739] [Core Development] [answered]
I am trying to debug some new methods that I am adding inside modnetwork.c - how to get ESP_LOGD or printf to print to console or is there another way to get debug statements printed from C code?

**Comment by jimmo** [comment, marked as answer]
`mp_printf(&mp_plat_print, "hello %d", x);`

**Reply by jimmo** [reply]
In most cases just straight printf will work too.

**Reply by krishnak** [reply]
printf didn't work

**Reply by russhughes** [reply]
Adding a fflush(stdout); after using printf() seems to help when debugging something that crashes soon after the printf(), mostly, but not always.

**Reply by krishnak** [reply]
thank you
